src/core/pluralize.js

```diff
@@ -35,10 +41,20 @@ export function run(conf) {
         dfn.dataset.plurals = uniquePlurals.join("|");
         registerDefinition(dfn, uniquePlurals);
       }
+      if (singulars.length) {
+        const userDefinedPlurals = dfn.dataset.singulars
+          ? dfn.dataset.singulars.split("|")
+          : [];
+        const uniqueSingulars = [
+          ...new Set([...userDefinedPlurals, ...singulars]),
+        ];
+        dfn.dataset.singulars = uniqueSingulars.join("|");
```

Comment:
We would prefer to not add new data attributes, so lets merge `uniquePlurals` and `uniqueSingulars` into `uniquePlurals` itself (and add to `dfn.dataset.plurals`)

Reply:
This isn't required if we use a single function as you suggested.